### PR TITLE
Unification Asclepios

### PR DESCRIPTION
**Pull request by Florent2305**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #
 ################################################################################
 
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.10)
```
Comment (mathildemerle) [inline CMakeLists.txt:14]: is this ok to change the pre-requisites? I'm on cmake 3.10 on Ubuntu 18, but we may have dev on version below

Comment (Florent2305) [reply CMakeLists.txt:14]: I Think it's a mistake on my part

File: src/plugins/legacy/itkINRDataImageReader/COPYING.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Copyright (c) <YEAR>, <OWNER>
```
Comment (mathildemerle) [inline src/plugins/legacy/itkINRDataImageReader/COPYING.txt:1]: is this file necessary? It could be removed, or the tags could be filled. Idem for README.txt

File: src/plugins/CMakeLists.txt [hunk 3]
```diff
@@ -49,6 +49,17 @@ set(PLUGIN_LIST
   legacy/itkDataSHImage                                                  ON
   legacy/undoRedoRegistration                                            ON
   legacy/diffeomorphicDemons                                             ON
+  
+  #legacy/itkDWIBrainMaskCalculatorProcess                                ON
+  legacy/itkINRDataImageReader                                           ON
+  legacy/itkINRDataImageWriter                                           ON
+  legacy/itkProcessRegistrationIncompressibleLogDemons                   ON
+  #legacy/itkProcessTensorLogEuclidean_legacy                             ON
```
Comment (ocommowi) [inline src/plugins/CMakeLists.txt:57]: Les vieux plugins non utilisés, c'est pas la peine de les garder en commentaire :)

File: src/plugins/legacy/itkINRDataImageReader/COPYING.txt (hunk 2 above)
Comment (ocommowi) [reply src/plugins/legacy/itkINRDataImageReader/COPYING.txt:1]: agreed, should be removed, we already have this at the root folder or so

File: src/plugins/legacy/itkINRDataImageWriter/COPYING.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Copyright (c) <YEAR>, <OWNER>
```
Comment (ocommowi) [inline src/plugins/legacy/itkINRDataImageWriter/COPYING.txt:1]: Same as before, remove

Comment (Florent2305) [reply src/plugins/legacy/itkINRDataImageWriter/COPYING.txt]: done

File: src/plugins/CMakeLists.txt (hunk 3 above)
Comment (Florent2305) [reply src/plugins/CMakeLists.txt]: c'était bien prévue
done

File: CMakeLists.txt (hunk 1 above)
Comment (Florent2305) [reply CMakeLists.txt:14]: done

File: src/plugins/legacy/itkINRDataImageReader/COPYING.txt (hunk 2 above)
Comment (Florent2305) [reply src/plugins/legacy/itkINRDataImageReader/COPYING.txt]: done

**Comment by Florent2305** [comment]
@ocommowi @msermesant Please! Before merging, please review and merge TTK and  LCC-LogDemons PRs

**Comment by msermesant** [comment]
I looked at the code, it seems fine to me, but I would rather let Olivier click on the big green button :-)

File: src/plugins/legacy/itkINRDataImageReader/itkINRDataImageReaderPlugin.doxyfile.in [hunk 5]
```diff
@@ -0,0 +1,1517 @@
+## #################################################################
```
Comment (mathildemerle) [inline src/plugins/legacy/itkINRDataImageReader/itkINRDataImageReaderPlugin.doxyfile.in:1]: to remove

File: src/layers/legacy/medImageIO/medInrimageImageIO.h [hunk 6]
```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+/**
```
Comment (ocommowi) [inline src/layers/legacy/medImageIO/medInrimageImageIO.h:3]: Unless Max is against it, change to medinria usual license text

File: src/layers/legacy/medImageIO/medInrimageImageIO.cpp [hunk 7]
```diff
@@ -0,0 +1,1172 @@
...
+    }
+    else
+    {
+        switch (m_PixelType)
+        {
+            case CHAR:
+                itk::ByteSwapper<char>::SwapRangeFromSystemToBigEndian((char *)buffer,
+                                                                  numberOfPixels);
+                break;
+            case UCHAR:
```
Comment (ocommowi) [inline src/layers/legacy/medImageIO/medInrimageImageIO.cpp:1130]: I get many warnings like this: /Users/ocommowi/medinria/medInria-public/src/layers/legacy/medImageIO/medInrimageImageIO.cpp:1130:18: warning: comparison of two values with different enumeration types in switch statement ('itk::ImageIOBase::IOPixelType' and 'itk::ImageIOBase::IOComponentType')

30 of them in the file, that's a lot. And I think it is actually ugly. To investigate

File: src/plugins/legacy/itkDataSHImage/interactors/itkDataSHImageVtkViewInteractor.cpp [hunk 8]
```diff
@@ -263,7 +263,6 @@ void itkDataSHImageVtkViewInteractor::setupParameters()
     medBoolParameterL *flipXParam = new medBoolParameterL("FlipX", this);
     medBoolParameterL *flipYParam = new medBoolParameterL("FlipY", this);
     medBoolParameterL *flipZParam = new medBoolParameterL("FlipZ", this);
-    flipZParam->setValue(true);
```
Comment (ocommowi) [inline src/plugins/legacy/itkDataSHImage/interactors/itkDataSHImageVtkViewInteractor.cpp:266]: My PR on SH changes was not applied to 3.1.x ? It's weird these modifications appear in this PR

File: src/plugins/legacy/itkINRDataImageReader/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+target_link_libraries(${TARGET_NAME}
+  Qt5::Core
+  ITKCommon
+  ${ITKIO_LIBRARIES}
+  medImageIO
+  dtkCoreSupport
+  dtkLog
+  medCoreLegacy)
+  #InrimageNG
```
Comment (ocommowi) [inline src/plugins/legacy/itkINRDataImageReader/CMakeLists.txt:98]: Remove these comments

File: src/plugins/legacy/itkINRDataImageWriter/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+target_link_libraries(${TARGET_NAME}
+  Qt5::Core
+  ITKCommon
+  ${ITKIO_LIBRARIES}
+  medImageIO
+  dtkCoreSupport
+  dtkLog
+  medCoreLegacy)
+  #InrimageNG
```
Comment (ocommowi) [inline src/plugins/legacy/itkINRDataImageWriter/CMakeLists.txt:98]: Remove these

File: src/plugins/legacy/itkINRDataImageReader/CMakeLists.txt (hunk 9 above)
Comment (Florent2305) [reply src/plugins/legacy/itkINRDataImageReader/CMakeLists.txt:98]: ok

File: src/plugins/legacy/itkINRDataImageWriter/CMakeLists.txt (hunk 10 above)
Comment (Florent2305) [reply src/plugins/legacy/itkINRDataImageWriter/CMakeLists.txt:98]: ok

**Comment by ocommowi** [comment]
Alright, good to go !! Starting by the other dependent branches
